src/ts/components/seekbarcontroller.ts

```diff
@@ -0,0 +1,111 @@
+import { UIUtils } from '../uiutils';
+import { DOM } from '../dom';
+import { PlayerAPI } from 'bitmovin-player';
+import { VolumeController } from '../volumecontroller';
+
+export enum SeekBarType {
+    Vod,
+    Live,
+    Volume,
+}
+
+const changeRangeValue = (
+    value: number,
+    range: { min: number, max: number },
+    cb: (value: number) => void,
+) => {
+    if (value < range.min) {
+        cb(range.min);
+    } else if (value > range.max) {
+        cb(range.max);
+    } else {
+        cb(value);
+    }
+};
+
+export class SeekBarController {
+    protected keyStepIncrements: { leftRight: number, upDown: number };
```

Comment:
It maybe would make sense to extract `{ leftRight: number, upDown: number }` into a `KeyStepIncrementsConfig` interface (or with a similar name).